pyresample/bucket/__init__.py

```diff
@@ -209,27 +208,47 @@ def _call_pandas_groupby_statistics(self, scipy_method, data, fill_value=None, s
         # Calculate the min of the data falling to each bin
         out_size = self.target_area.size
 
-        # merge into one Dataframe
-        df = dd.concat([dd.from_dask_array(self.idxs), dd.from_dask_array(weights)],
-                       axis=1)
-        df.columns = ['x', 'values']
+        # create the output bins
+        bins = da.linspace(0, out_size-1, out_size).astype('int')
 
-        if scipy_method == 'min':
-            statistics = df.map_partitions(lambda part: part.groupby(
-                                           np.digitize(part.x,
-                                                       bins=np.linspace(0, out_size, out_size)
-                                                       )
-                                           )['values'].min())
+        # sort idxs and weights based on weights
+        if statistic_method == 'min':
+            order = np.argsort(weights)
```

Comment:
`weights` is a dask array that is based on the input `data`. By doing `np.argsort` here you are likely computing the whole data array here and then again later when you get the final output. Can this be moved into one of the delayed functions?